core/src/main/java/com/google/errorprone/bugpatterns/ModifiedButNotUsed.java

```diff
@@ -286,10 +287,11 @@ public Description matchExpressionStatement(ExpressionStatementTree tree, Visito
    * Whether this is a chain of method invocations terminating in a new proto or collection builder.
    */
   private static boolean newFluentChain(ExpressionTree tree, VisitorState state) {
-    while (tree instanceof MethodInvocationTree && FLUENT_CHAIN.matches(tree, state)) {
-      tree = getReceiver(tree);
-    }
-    return tree != null && FLUENT_CONSTRUCTOR.matches(tree, state);
+    return concat(Stream.of(tree), streamReceivers(tree))
+        .filter(t -> !FLUENT_CHAIN.matches(t, state))
+        .findFirst()
+        .map(t -> FLUENT_CONSTRUCTOR.matches(t, state))
+        .orElse(false);
```

Comment:
Without boxing:
```suggestion
        .filter(t -> FLUENT_CONSTRUCTOR.matches(t, state))
        .isPresent();
```

Reply:
I don't really love that alternative: it avoids boxing, but I think `isPresent` expresses the intent a bit less clearly.

(Mostly I find it weird almost ever using `ifPresent` without `get`.)